runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -309,33 +294,14 @@ else if (event.getMenuAction() != MenuAction.GAME_OBJECT_FIRST_OPTION)
 	}
 
 	@Subscribe
-	public void onProjectileMoved(ProjectileMoved event)
+	public void onVarbitChanged(VarbitChanged varbitChanged)
 	{
-		Projectile projectile = event.getProjectile();
+		cballsLeft = client.getVarpValue(VarPlayer.CANNON_AMMO.getId());
 
-		if (CANNONBALL_PROJECTILE_IDS.contains(projectile.getId()) && cannonPosition != null && cannonWorld == client.getWorld())
+		if (config.showCannonNotifications() && !cannonBallNotificationSent && cballsLeft > 0 && config.lowWarningThreshold() >= cballsLeft)
```

Comment:
Either check the event index or otherwise avoid hitting a config call per varbit change

Reply:
I've made it so that the varp value is only obtained if the index matches the CANNON_AMMO id. Thank you for the feedback